File: merry360x_mobile/ios-app/Merry360xMobile/Views/AuthBottomSheet.swift
import SwiftUI

enum AuthMode: String, CaseIterable {
    case login = "Log in"
    case signup = "Sign up"
}

enum AuthInputTab: String, CaseIterable {
    case email = "Email"
    case phone = "Phone"
}

struct AuthBottomSheet: View {
    @EnvironmentObject private var session: AppSessionViewModel
    @Environment(\.openURL) private var openURL
    @StateObject private var viewModel = AuthViewModel()
    @Binding var isPresented: Bool
    
    @State private var authMode: AuthMode = .login
    @State private var inputTab: AuthInputTab = .email
    @State private var phone = ""
    @State private var confirmPassword = ""
    @State private var name = ""
    @State private var showPassword = false
    
    var body: some View {
        VStack(spacing: 0) {
            // Drag Handle and Close
            HStack {
                Spacer()
                    .frame(width: 44)
                
                Spacer()
                
                RoundedRectangle(cornerRadius: 2)
                    .fill(Color.gray.opacity(0.4))
                    .frame(width: 40, height: 4)
                
                Spacer()
                
                Button(action: { isPresented = false }) {
                    Image(systemName: "xmark")
                        .font(.system(size: 16, weight: .medium))
                        .foregroundColor(.gray)
                        .frame(width: 44, height: 44)
                }
            }
            .padding(.top, 8)
            
            ScrollView {
                VStack(spacing: 20) {
                    // Logo
                    ZStack {
                        Circle()
                            .fill(AppTheme.coral)
                            .frame(width: 64, height: 64)
                        
                        Text("M")
                            .font(.system(size: 28, weight: .bold))
                            .foregroundColor(.white)
                    }
                    
                    // Title
                    VStack(spacing: 4) {
                        Text(authMode == .login ? "Welcome back" : "Create account")
                            .font(.system(size: 24, weight: .bold))
                        
                        Text(authMode == .login ? "Sign in to continue" : "Join Merry360x today")
                            .font(.system(size: 14))
                            .foregroundColor(AppTheme.textSecondary)
                    }
                    
                    // Login / Signup Toggle
                    HStack(spacing: 4) {
                        ForEach(AuthMode.allCases, id: \.self) { mode in
                            Button(action: {
                                withAnimation(.easeInOut(duration: 0.2)) {
                                    authMode = mode
                                }
                            }) {
                                Text(mode.rawValue)
                                    .font(.system(size: 14, weight: authMode == mode ? .semibold : .regular))
                                    .foregroundColor(authMode == mode ? .black : .gray)
                                    .frame(maxWidth: .infinity)
                                    .padding(.vertical, 10)
                                    .background(authMode == mode ? Color.white : Color.clear)
                                    .clipShape(RoundedRectangle(cornerRadius: 8))
                            }
                        }
                    }
                    .padding(4)
                    .background(AppTheme.cardBackground)
                    .clipShape(RoundedRectangle(cornerRadius: 12))
                    
                    // Email / Phone Tab
                    HStack(spacing: 0) {
                        ForEach(AuthInputTab.allCases, id: \.self) { tab in
                            Button(action: {
                                withAnimation(.easeInOut(duration: 0.2)) {
                                    inputTab = tab
                                }
                            }) {
                                VStack(spacing: 8) {
                                    HStack(spacing: 6) {
                                        Image(systemName: tab == .email ? "envelope" : "phone")
                                            .font(.system(size: 14))
                                        Text(tab.rawValue)
                                            .font(.system(size: 14, weight: inputTab == tab ? .semibold : .regular))
                                    }
                                    .foregroundColor(inputTab == tab ? AppTheme.coral : .gray)
                                    
                                    Rectangle()
                                        .fill(inputTab == tab ? AppTheme.coral : Color.clear)
                                        .frame(height: 2)
                                }
                            }
                            .frame(maxWidth: .infinity)
                        }
                    }
                    
                    // Form Fields
                    VStack(spacing: 12) {
                        // Name (signup only)
                        if authMode == .signup {
                            AuthTextField(
                                label: "Full Name",
                                placeholder: "Enter your name",
                                text: $name
                            )
                        }
                        
                        // Email or Phone
                        if inputTab == .email {
                            AuthTextField(
                                label: "Email",
                                placeholder: "Enter your email",
                                text: $viewModel.email,
                                keyboardType: .emailAddress,
                                autocapitalization: .never
                            )
                        } else {
                            HStack(spacing: 8) {
                                // Country code
                                HStack {
                                    Text("🇷🇼")
                                    Text("+250")
                                        .foregroundColor(AppTheme.textPrimary)
                                }
                                .padding(.horizontal, 12)
                                .padding(.vertical, 14)
                                .background(AppTheme.cardBackground)
                                .clipShape(RoundedRectangle(cornerRadius: 12))
                                .overlay(
                                    RoundedRectangle(cornerRadius: 12)
                                        .stroke(Color.gray.opacity(0.3), lineWidth: 1)
                                )
                                
                                AuthTextField(
                                    label: "",
                                    placeholder: "Phone number",
                                    text: $phone,
                                    keyboardType: .phonePad
                                )
                            }
                        }
                        
                        // Password
                        AuthSecureField(
                            label: "Password",
                            placeholder: "Enter your password",
                            text: $viewModel.password,
                            showPassword: $showPassword
                        )
                        
                        // Confirm Password (signup only)
                        if authMode == .signup {
                            AuthSecureField(
                                label: "Confirm Password",
                                placeholder: "Confirm your password",
                                text: $confirmPassword,
                                showPassword: .constant(false)
                            )
                        }
                        
                        // Forgot Password (login only)
                        if authMode == .login {
                            HStack {
                                Spacer()
                                Button("Forgot Password?") {
                                    // Handle forgot password
                                }
                                .font(.system(size: 13))
                                .foregroundColor(AppTheme.coral)
                            }
                        }
                    }
                    
                    // Error Message
                    if let error = viewModel.errorMessage {
                        Text(error)
                            .font(.system(size: 13))
                            .foregroundColor(.red)
                            .multilineTextAlignment(.center)
                    }
                    
                    // Main Action Button
                    Button(action: handleAuth) {
                        HStack {
                            if viewModel.loading {
                                ProgressView()
                                    .tint(.white)
                            } else {
                                Text(authMode == .login ? "Sign In" : "Create Account")
                                    .font(.system(size: 16, weight: .semibold))
                            }
                        }
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 16)
                        .background(AppTheme.coral)
                        .clipShape(RoundedRectangle(cornerRadius: 12))
                    }
                    .disabled(viewModel.loading)
                    
                    // Divider
                    HStack {
                        Rectangle()
                            .fill(Color.gray.opacity(0.3))
                            .frame(height: 1)
                        
                        Text("or continue with")
                            .font(.system(size: 13))
                            .foregroundColor(AppTheme.textSecondary)
                        
                        Rectangle()
                            .fill(Color.gray.opacity(0.3))
                            .frame(height: 1)
                    }
                    
                    // Social Login Buttons
                    HStack(spacing: 12) {
                        // Apple
                        Button(action: { handleOAuth(provider: "apple") }) {
                            HStack(spacing: 8) {
                                Image(systemName: "apple.logo")
                                    .font(.system(size: 18))
                                Text("Apple")
                                    .font(.system(size: 15, weight: .medium))
                            }
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 14)
                            .background(Color.black)
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                        }
                        
                        // Google
                        Button(action: { handleOAuth(provider: "google") }) {
                            HStack(spacing: 8) {
                                Text("G")
                                    .font(.system(size: 18, weight: .bold))
                                    .foregroundColor(.red)
                                Text("Google")
                                    .font(.system(size: 15, weight: .medium))
                                    .foregroundColor(.black)
                            }
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 14)
                            .background(Color.white)
                            .overlay(
                                RoundedRectangle(cornerRadius: 12)
                                    .stroke(Color.gray.opacity(0.3), lineWidth: 1)
                            )
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                        }
                    }
                    
                    // Terms
                    Text("By continuing, you agree to our Terms of Service and Privacy Policy")
                        .font(.system(size: 11))
                        .foregroundColor(AppTheme.textSecondary)
                        .multilineTextAlignment(.center)
                        .padding(.top, 8)
                }
                .padding(.horizontal, 24)
                .padding(.bottom, 32)
            }
        }
        .background(Color.white)
    }
    
    private func handleAuth() {
        guard !viewModel.loading else { return }
        
        // Set email from phone if using phone tab
        if inputTab == .phone {
            viewModel.email = "+250\(phone)"
        }
        
        guard !viewModel.email.isEmpty, !viewModel.password.isEmpty else {
            viewModel.errorMessage = "Please fill in all fields"
            return
        }
        
        if authMode == .signup {
            guard !name.isEmpty else {
                viewModel.errorMessage = "Please enter your name"
                return
            }
            guard viewModel.password == confirmPassword else {
                viewModel.errorMessage = "Passwords do not match"
                return
            }
        }
        
        Task {
            if authMode == .signup {
                await viewModel.signUp(session: session)
            } else {
                await viewModel.signIn(session: session)
            }
            if session.isAuthenticated {
                isPresented = false
            }
        }
    }

    private func handleOAuth(provider: String) {
        guard let service = SupabaseService(), let url = service.oauthAuthorizeURL(provider: provider) else {
            viewModel.errorMessage = "Unable to start \(provider.capitalized) sign in."
            return
        }
        openURL(url)
    }
}

// MARK: - Text Field Components

struct AuthTextField: View {
    let label: String
    let placeholder: String
    @Binding var text: String
    var keyboardType: UIKeyboardType = .default
    var autocapitalization: TextInputAutocapitalization = .sentences
    
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            if !label.isEmpty {
                Text(label)
                    .font(.system(size: 13, weight: .medium))
                    .foregroundColor(.gray)
            }
            
            TextField(placeholder, text: $text)
                .keyboardType(keyboardType)
                .textInputAutocapitalization(autocapitalization)
                .padding(.horizontal, 16)
                .padding(.vertical, 14)
                .background(AppTheme.cardBackground)
                .clipShape(RoundedRectangle(cornerRadius: 12))
                .overlay(
                    RoundedRectangle(cornerRadius: 12)
                        .stroke(Color.gray.opacity(0.3), lineWidth: 1)
                )
        }
    }
}

struct AuthSecureField: View {
    let label: String
    let placeholder: String
    @Binding var text: String
    @Binding var showPassword: Bool
    
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(label)
                .font(.system(size: 13, weight: .medium))
                .foregroundColor(.gray)
            
            HStack {
                if showPassword {
                    TextField(placeholder, text: $text)
                        .textContentType(.password)
                } else {
                    SecureField(placeholder, text: $text)
                        .textContentType(.password)
                }
                
                Button(action: { showPassword.toggle() }) {
                    Image(systemName: showPassword ? "eye.slash" : "eye")
                        .foregroundColor(.gray)
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 14)
            .background(AppTheme.cardBackground)
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(Color.gray.opacity(0.3), lineWidth: 1)
            )
        }
    }
}

#Preview {
    AuthBottomSheet(isPresented: .constant(true))
        .environmentObject(AppSessionViewModel())
}

File: merry360x_mobile/android-app/app/src/main/java/com/merry360x/mobile/ui/screens/AuthBottomSheet.kt
package com.merry360x.mobile.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Email
import androidx.compose.material.icons.filled.Phone
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.SheetState
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.merry360x.mobile.theme.CardGray
import com.merry360x.mobile.theme.Coral
import com.merry360x.mobile.viewmodel.AuthUiState

private enum class SheetAuthMode {
    LOGIN, SIGNUP
}

private enum class SheetAuthTab {
    EMAIL, PHONE
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AuthBottomSheet(
    showSheet: Boolean,
    sheetState: SheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true),
    uiState: AuthUiState,
    onEmail: (String) -> Unit,
    onPassword: (String) -> Unit,
    onPhone: (String) -> Unit = {},
    onSignIn: () -> Unit,
    onSignUp: () -> Unit = {},
    onSignInWithApple: () -> Unit = {},
    onSignInWithGoogle: () -> Unit = {},
    onForgotPassword: () -> Unit = {},
    onDismiss: () -> Unit
) {
    var authMode by remember { mutableStateOf(SheetAuthMode.LOGIN) }
    var selectedTab by remember { mutableStateOf(SheetAuthTab.EMAIL) }
    var passwordVisible by remember { mutableStateOf(false) }
    var phoneNumber by remember { mutableStateOf("") }
    var confirmPassword by remember { mutableStateOf("") }
    var name by remember { mutableStateOf("") }
    
    if (showSheet) {
        ModalBottomSheet(
            onDismissRequest = onDismiss,
            sheetState = sheetState,
            containerColor = Color.White,
            shape = RoundedCornerShape(topStart = 24.dp, topEnd = 24.dp),
            dragHandle = null
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 24.dp)
                    .padding(bottom = 32.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Handle bar and close button
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(top = 12.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Spacer(modifier = Modifier.width(40.dp))
                    
                    // Drag handle
                    Box(
                        modifier = Modifier
                            .width(40.dp)
                            .height(4.dp)
                            .clip(RoundedCornerShape(2.dp))
                            .background(Color(0xFFDDDDDD))
                    )
                    
                    IconButton(onClick = onDismiss) {
                        Icon(
                            imageVector = Icons.Default.Close,
                            contentDescription = "Close",
                            tint = Color(0xFF9E9E9E)
                        )
                    }
                }
                
                Spacer(modifier = Modifier.height(16.dp))
                
                // Logo
                Box(
                    modifier = Modifier
                        .size(64.dp)
                        .clip(CircleShape)
                        .background(Coral),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "M",
                        fontSize = 28.sp,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )
                }
                
                Spacer(modifier = Modifier.height(16.dp))
                
                // Title
                Text(
                    text = if (authMode == SheetAuthMode.LOGIN) "Welcome back" else "Create account",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
                
                Text(
                    text = if (authMode == SheetAuthMode.LOGIN) "Sign in to continue" else "Join Merry360x today",
                    fontSize = 14.sp,
                    color = Color(0xFF9E9E9E)
                )
                
                Spacer(modifier = Modifier.height(24.dp))
                
                // Login / Sign Up Toggle
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(12.dp))
                        .background(CardGray)
                        .padding(4.dp),
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    SheetAuthMode.values().forEach { mode ->
                        val isSelected = authMode == mode
                        Box(
                            modifier = Modifier
                                .weight(1f)
                                .clip(RoundedCornerShape(8.dp))
                                .background(if (isSelected) Color.White else Color.Transparent)
                                .clickable { authMode = mode }
                                .padding(vertical = 10.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(
                                text = if (mode == SheetAuthMode.LOGIN) "Log in" else "Sign up",
                                fontSize = 14.sp,
                                fontWeight = if (isSelected) FontWeight.SemiBold else FontWeight.Normal,
                                color = if (isSelected) Color.Black else Color(0xFF9E9E9E)
                            )
                        }
                    }
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Email / Phone Tab Selector
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.Center
                ) {
                    SheetAuthTab.values().forEach { tab ->
                        val isSelected = selectedTab == tab
                        Column(
                            modifier = Modifier
                                .weight(1f)
                                .clickable { selectedTab = tab },
                            horizontalAlignment = Alignment.CenterHorizontally
                        ) {
                            Row(
                                verticalAlignment = Alignment.CenterVertically,
                                horizontalArrangement = Arrangement.Center
                            ) {
                                Icon(
                                    imageVector = if (tab == SheetAuthTab.EMAIL) Icons.Default.Email else Icons.Default.Phone,
                                    contentDescription = null,
                                    tint = if (isSelected) Coral else Color(0xFF9E9E9E),
                                    modifier = Modifier.size(18.dp)
                                )
                                Spacer(modifier = Modifier.width(6.dp))
                                Text(
                                    text = if (tab == SheetAuthTab.EMAIL) "Email" else "Phone",
                                    fontSize = 14.sp,
                                    fontWeight = if (isSelected) FontWeight.SemiBold else FontWeight.Normal,
                                    color = if (isSelected) Coral else Color(0xFF9E9E9E)
                                )
                            }
                            Spacer(modifier = Modifier.height(8.dp))
                            Box(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .height(2.dp)
                                    .background(if (isSelected) Coral else Color.Transparent)
                            )
                        }
                    }
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Name field (only for signup)
                if (authMode == SheetAuthMode.SIGNUP) {
                    OutlinedTextField(
                        value = name,
                        onValueChange = { name = it },
                        label = { Text("Full Name") },
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Coral,
                            unfocusedBorderColor = Color(0xFFDDDDDD),
                            focusedContainerColor = CardGray,
                            unfocusedContainerColor = CardGray
                        ),
                        singleLine = true
                    )
                    Spacer(modifier = Modifier.height(12.dp))
                }
                
                // Email or Phone Input
                if (selectedTab == SheetAuthTab.EMAIL) {
                    OutlinedTextField(
                        value = uiState.email,
                        onValueChange = onEmail,
                        label = { Text("Email address") },
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Coral,
                            unfocusedBorderColor = Color(0xFFDDDDDD),
                            focusedContainerColor = CardGray,
                            unfocusedContainerColor = CardGray
                        ),
                        singleLine = true
                    )
                } else {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        // Country code
                        OutlinedTextField(
                            value = "+250",
                            onValueChange = {},
                            modifier = Modifier.width(90.dp),
                            shape = RoundedCornerShape(12.dp),
                            colors = OutlinedTextFieldDefaults.colors(
                                focusedBorderColor = Coral,
                                unfocusedBorderColor = Color(0xFFDDDDDD),
                                focusedContainerColor = CardGray,
                                unfocusedContainerColor = CardGray
                            ),
                            readOnly = true,
                            leadingIcon = { Text("🇷🇼") },
                            singleLine = true
                        )
                        
                        // Phone number
                        OutlinedTextField(
                            value = phoneNumber,
                            onValueChange = { 
                                phoneNumber = it
                                onPhone("+250$it")
                            },
                            label = { Text("Phone number") },
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(12.dp),
                            colors = OutlinedTextFieldDefaults.colors(
                                focusedBorderColor = Coral,
                                unfocusedBorderColor = Color(0xFFDDDDDD),
                                focusedContainerColor = CardGray,
                                unfocusedContainerColor = CardGray
                            ),
                            singleLine = true
                        )
                    }
                }
                
                Spacer(modifier = Modifier.height(12.dp))
                
                // Password Field
                OutlinedTextField(
                    value = uiState.password,
                    onValueChange = onPassword,
                    label = { Text("Password") },
                    modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(12.dp),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedBorderColor = Coral,
                        unfocusedBorderColor = Color(0xFFDDDDDD),
                        focusedContainerColor = CardGray,
                        unfocusedContainerColor = CardGray
                    ),
                    visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
                    trailingIcon = {
                        IconButton(onClick = { passwordVisible = !passwordVisible }) {
                            Icon(
                                imageVector = if (passwordVisible) Icons.Default.Visibility else Icons.Default.VisibilityOff,
                                contentDescription = if (passwordVisible) "Hide password" else "Show password",
                                tint = Color(0xFF9E9E9E)
                            )
                        }
                    },
                    singleLine = true
                )
                
                // Confirm Password (only for signup)
                if (authMode == SheetAuthMode.SIGNUP) {
                    Spacer(modifier = Modifier.height(12.dp))
                    OutlinedTextField(
                        value = confirmPassword,
                        onValueChange = { confirmPassword = it },
                        label = { Text("Confirm Password") },
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Coral,
                            unfocusedBorderColor = Color(0xFFDDDDDD),
                            focusedContainerColor = CardGray,
                            unfocusedContainerColor = CardGray
                        ),
                        visualTransformation = PasswordVisualTransformation(),
                        singleLine = true
                    )
                }
                
                // Forgot Password (only for login)
                if (authMode == SheetAuthMode.LOGIN) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.End
                    ) {
                        Text(
                            text = "Forgot Password?",
                            fontSize = 13.sp,
                            color = Coral,
                            modifier = Modifier.clickable { onForgotPassword() }
                        )
                    }
                }
                
                // Error message
                if (uiState.error != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(
                        text = uiState.error,
                        fontSize = 13.sp,
                        color = Color.Red,
                        textAlign = TextAlign.Center,
                        modifier = Modifier.fillMaxWidth()
                    )
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Main Action Button
                Button(
                    onClick = { if (authMode == SheetAuthMode.LOGIN) onSignIn() else onSignUp() },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(52.dp),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Coral),
                    enabled = !uiState.loading
                ) {
                    Text(
                        text = if (uiState.loading) "Please wait..." 
                               else if (authMode == SheetAuthMode.LOGIN) "Sign In" 
                               else "Create Account",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.SemiBold
                    )
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Divider with "or"
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    HorizontalDivider(
                        modifier = Modifier.weight(1f),
                        color = Color(0xFFDDDDDD)
                    )
                    Text(
                        text = "  or continue with  ",
                        fontSize = 13.sp,
                        color = Color(0xFF9E9E9E)
                    )
                    HorizontalDivider(
                        modifier = Modifier.weight(1f),
                        color = Color(0xFFDDDDDD)
                    )
                }
                
                Spacer(modifier = Modifier.height(20.dp))
                
                // Social Login Buttons
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    // Apple
                    Box(
                        modifier = Modifier
                            .weight(1f)
                            .height(52.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .background(Color.Black)
                            .clickable { onSignInWithApple() },
                        contentAlignment = Alignment.Center
                    ) {
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.Center
                        ) {
                            Text("", fontSize = 20.sp) // Apple logo emoji
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = "Apple",
                                color = Color.White,
                                fontSize = 15.sp,
                                fontWeight = FontWeight.Medium
                            )
                        }
                    }
                    
                    // Google
                    Box(
                        modifier = Modifier
                            .weight(1f)
                            .height(52.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .border(1.dp, Color(0xFFDDDDDD), RoundedCornerShape(12.dp))
                            .background(Color.White)
                            .clickable { onSignInWithGoogle() },
                        contentAlignment = Alignment.Center
                    ) {
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.Center
                        ) {
                            Text(
                                text = "G",
                                fontSize = 20.sp,
                                fontWeight = FontWeight.Bold,
                                color = Color.Red
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = "Google",
                                color = Color.Black,
                                fontSize = 15.sp,
                                fontWeight = FontWeight.Medium
                            )
                        }
                    }
                }
                
                Spacer(modifier = Modifier.height(16.dp))
                
                // Terms text
                Text(
                    text = "By continuing, you agree to our Terms of Service and Privacy Policy",
                    fontSize = 11.sp,
                    color = Color(0xFF9E9E9E),
                    textAlign = TextAlign.Center,
                    modifier = Modifier.fillMaxWidth()
                )
            }
        }
    }
}
